src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -943,62 +984,16 @@ internal virtual void PrepareSession()
                 WebSession.RetryIntervalInSeconds = RetryIntervalSec;
             }
         }
-        
+
         internal virtual HttpClient GetHttpClient(bool handleRedirect)
         {
-            HttpClientHandler handler = new();
-            handler.CookieContainer = WebSession.Cookies;
-            handler.AutomaticDecompression = DecompressionMethods.All;
 
-            // Set the credentials used by this request
-            if (WebSession.UseDefaultCredentials)
+            var client = WebSession.GetHttpClient(handleRedirect, TimeoutSec, out var newClient);
+            if (newClient && IsPersistentSession() && SessionVariable is null)
             {
-                // The UseDefaultCredentials flag overrides other supplied credentials
-                handler.UseDefaultCredentials = true;
+                WriteWarning("WebSession properties changed: new Http connection will be required");
```

Comment:
It seems the condition is wrong.

Reply:
I don't know what you're referring to here. Could you provide more detail? If it is related to the tests failing, I have identified that as being caused by changing the MaxAutomaticRedirections property on the WebSession while processing the CmdLet, which I had not realised was occurring. This caused the WebRequestSession to Dispose of the in-use client.

This should be resolved in my latest commit, which includes the changes by @CarloToso from #19190 as well as additional changes suggested to remove the property entirely from the WebRequestSession.